src/clj/puppetlabs/services/master/master_service.clj

```diff
@@ -165,9 +173,18 @@
       "master"
       (status-core/get-artifact-version "puppetlabs" "puppetserver")
       master-service-status-version
-      (partial core/v1-status metrics))
-     (assoc context :http-metrics metrics)))
+      (partial core/v1-status http-metrics http-client-metric-ids-for-status registry))
+     (-> context
+         (assoc :http-metrics http-metrics)
+         (assoc :http-client-metric-ids-for-status http-client-metric-ids-for-status))))
   (start
     [this context]
     (log/info (i18n/trs "Puppet Server has successfully started and is now ready to handle requests"))
-    context))
+    context)
+
+  (add-metric-ids-to-http-client-metrics-list!
+   [this metric-ids-to-add]
+   (let [metric-ids-from-context (:http-client-metric-ids-for-status
+                                  (tk-services/service-context this))]
+     (master-core/add-metric-ids-to-http-client-metrics-list! metric-ids-from-context
```

Comment:
Do you think it's worth "locking" the list so that nothing can append to it after the service has started, like we do with registries in tk-metrics?  I think I'm good with not adding that on - just curious if you'd thought about it.